### PR TITLE
🔧 vue3 composition api

### PR DESCRIPTION
**Pull request by roiLeo**
- part of #4750 
- `EmotionList` (not used anymore)
- `History` Big component
Screenshot 2023-08-14 at 11-55-38 zinc KodaDot - NFT Market Explorer
- User profile Sales 
- `Holder` composition api

### CODE REVIEW

File: components/rmrk/Gallery/History.vue [hunk 1]
```diff
@@ -108,287 +102,242 @@ import {
...
+  return ''
+}
+const replaceUrl = (value: string, key = 'event') => {
+  if ($route.query[key] !== value) {
+    $router
+      .replace({
+        path: String($route.path),
+        query: { ...$route.query, [key]: value },
+      })
+      .catch($consola.warn /*Navigation Duplicate err fix later */)
```
Comment (daiagi) [inline components/rmrk/Gallery/History.vue:208]: we have a composable that does that

File: components/rmrk/Gallery/History.vue [hunk 2]
```diff
@@ -108,287 +102,242 @@ import {
...
   }
-
-  private percentageTextClassName(percentage: number) {
-    if (percentage > 0) {
-      return 'has-text-success'
-    } else if (percentage < 0) {
-      return 'has-text-danger'
+  const previousPriceMap = {}
+
+  console.log(prop.events)
```
Comment (daiagi) [inline components/rmrk/Gallery/History.vue:232]: looks like you forgot some console logs :see_no_evil:

File: components/rmrk/Gallery/History.vue [hunk 3]
```diff
@@ -108,287 +102,242 @@ import {
...
-    }
-  }
+  emit('setPriceChartData', [chartData.buy, chartData.list])
 }
+
+watch(
+  () => prop.events,
+  () => {
+    createTable()
+  }
```
Comment (daiagi) [inline components/rmrk/Gallery/History.vue:337]:
```suggestion
createTable
```


not important, just style preference, feel free to resolve and ignore if you disagree

File: components/rmrk/Gallery/Holder/Holder.vue [hunk 4]
```diff
@@ -232,320 +217,277 @@ type BaseTableRow = {
...
-    }
-  }
+const replaceUrl = (value: string, key = 'page') => {
+  $router
+    .replace({
+      path: String($route.path),
+      query: { ...$route.query, [key]: value },
+    })
+    .catch($consola.warn /*Navigation Duplicate err fix later */)
+}
```
Comment (daiagi) [inline components/rmrk/Gallery/Holder/Holder.vue:476]: as before, we have composable that does this

File: components/rmrk/Gallery/Holder/Holder.vue [hunk 5]
```diff
@@ -232,320 +217,277 @@ type BaseTableRow = {
...
 
-  @Watch('currentPage')
-  watchPageValue(val) {
-    this.replaceUrl(String(val))
+watch(
+  () => prop.events,
+  () => {
+    createTable()
   }
+)
```
Comment (daiagi) [inline components/rmrk/Gallery/Holder/Holder.vue:483]:
```suggestion
watch(
  () => prop.events,
  createTable
)
```

File: components/rmrk/Gallery/Holder/Holder.vue [hunk 6]
```diff
@@ -232,320 +217,277 @@ type BaseTableRow = {
...
-    this.$router
-      .replace({
-        path: String(this.$route.path),
-        query: { ...this.$route.query, [key]: value },
-      })
-      .catch(this.$consola.warn /*Navigation Duplicate err fix later */)
+watch(
+  () => prop.tableRowsOption,
+  () => {
+    createTableByTableRow()
```
Comment (daiagi) [inline components/rmrk/Gallery/Holder/Holder.vue:488]:
```suggestion
watch(
  () => prop.tableRowsOption,
  createTableByTableRow
  ```

**Review by daiagi — COMMENTED** [review]
smol stuff here and there

File: components/rmrk/Gallery/History.vue (hunk 1 above)
Comment (roiLeo) [reply components/rmrk/Gallery/History.vue:208]: Cool! I'll try to use it

File: components/bsx/Offer/OfferTable.vue [hunk 7]
```diff
@@ -188,21 +189,9 @@ const tellFrens = (caller: string, withdraw: boolean) => {
 }
 
 watch(currentPage, (val) => {
-  replaceUrl(String(val))
+  replaceUrl({ page: String(val) })
```
Comment (vikiival) [inline components/bsx/Offer/OfferTable.vue:192]: GIga cHad Enerygy

**Review by vikiival — COMMENTED** [review]
smol

File: components/rmrk/Gallery/History.vue [hunk 8]
```diff
@@ -74,7 +74,7 @@
         field="Percentage"
         label="Percentage">
         <span :class="percentageTextClassName(props.row.Percentage)">
-          {{ props.row.Percentage | toPercent('-') }}
+          {{ props.row.Percentage }}
```
[the remark this answers is not in the document]
Comment (roiLeo) [reply components/rmrk/Gallery/History.vue]: now using utils, idk why I removed this part

**Review by daiagi — APPROVED** [review]

**Pull request merged**